vllm/model_executor/tensorizer_loader.py

```diff
@@ -0,0 +1,249 @@
+import argparse
+import dataclasses
+import io
+import os
+import time
+import typing
+import warnings
+from dataclasses import dataclass
+from typing import Optional, Type, Union
+
+from tensorizer import DecryptionParams, TensorDeserializer, stream_io
+from tensorizer.utils import convert_bytes, get_mem_usage, no_init_or_tensor
+from torch import nn
+
+from vllm.config import ModelConfig
+from vllm.logger import init_logger
+
+logger = init_logger(__name__)
+
+
+def load_with_tensorizer(model_cls: Type[nn.Module],
+                         model_config: ModelConfig) -> nn.Module:
+    tensorizer = TensorizerAgent(model_cls, model_config)
+    return tensorizer.deserialize()
+
+
+def _is_vllm_model(model_config: ModelConfig = None,
+                   file_uri: Optional[str] = None) -> bool:
+    if file_uri:
+        return "vllm" in file_uri
+    else:
+        return "vllm" in model_config.tensorizer_args.tensorizer_uri
```

Comment:
this is pretty hacky and magical, I think it would be better to instead have users determine this through an argument (in `tensorizer_args`)